### DISCUSSION TITLE
mspms genome length limitation

### DISCUSSION THREAD

**Discussion by AlisaGU** [tskit-dev/msprime#1888] [Q&A]
Hey, It's me again!
My friend used mspms to simulate a genome  of 100M base pairs. After waiting a long time, the program is still running.
However, when he changed to python, program running quickly.
I am not sure what's the key.

**Comment by benjeffery** [comment]
Could you give us some more details, ideally the code snippet and command line?

**Comment by AlisaGU** [comment]
Sorry, I don't have these codes... Maybe my friend will give me a copy in tomorrow.
Empirically, will command line and python code run the same time ideally?

**Comment by grahamgower** [comment]
> Sorry, I don't have these codes... Maybe my friend will give me a copy in tomorrow. Empirically, will command line and python code run the same time ideally?

Well, `msp` will output a tree sequence, so this should run at a speed on par with using the python API to get a tree sequence (and then writing it to disk). Whereas `mspms` outputs the ms text format, which is likely to be slower. Without looking at the code/command in question, it's hard to guess if the text output is the cause of the slowness.

**Comment by grahamgower** [comment]
Possibly related to #1652.

**Comment by AlisaGU** [comment]
Thanks
Before this issue, I have read #1652。
I simulate a genome of 10M using mspms and ms. Mspms just took 3 minutes. It's much faster than ms. 
Now I am simulating 100M to observe what will happen.
